zcncore/wallet.go

```diff
@@ -445,11 +446,12 @@ func GetNetworkJSON() string {
 
 // CreateWallet creates the wallet for to configure signature scheme.
 // It also registers the wallet again to block chain.
-func CreateWallet(statusCb WalletCallback) error {
+func CreateWallet(wg *sync.WaitGroup, statusCb WalletCallback) error {
 	if len(_config.chain.Miners) < 1 || len(_config.chain.Sharders) < 1 {
 		return errors.New("", "SDK not initialized")
 	}
 	go func() {
+		defer wg.Done()
```

Comment:
it may not work properly. because wg.Done should already be called on statusCb.OnWalletCreateComplete.
it should be better to update WalletCallback with
```
// WalletCallback needs to be implemented for wallet creation.
type WalletCallback interface {
	OnWalletCreateComplete(status int, wallet string, err string)
        OnWalletCreateFailed(status int, wallet string, err string)
}
```

Reply:
I have changed that logic in both zwallet and zbox to reflect the changes here
to make it more as per golang language pattern expectation...